scholarly/publication.py

```diff
@@ -122,10 +134,13 @@ def _scholar_pub(self, __data):
         if title.find('a'):
             self.bib['url'] = title.find('a')['href']
 
-        authorinfo = databox.find('div', class_='gs_a').text
+        author_div_element = databox.find('div', class_='gs_a')
+        authorinfo = author_div_element.text
         authorinfo = authorinfo.replace(u'\xa0', u' ')       # NBSP
         authorinfo = authorinfo.replace(u'&amp;', u'&')      # Ampersand
         self.bib["author"] = self._get_authorlist(authorinfo)
+        authorinfo_html = author_div_element.decode_contents()
+        self.bib["author_id"] = self._get_author_id_list(authorinfo_html)
```

Comment:
returns an array that matches positionally with `author` field. author_id can be `None` if not linked in gs.